xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/internal/html/DefaultHTMLCleanerTest.java

```diff
@@ -661,18 +664,104 @@ public void parseWithUnicodeChars() throws Exception
+    @Test
+    void replaceOpeningBraceCharacterEntityByNamedEntity()
+    {
+        assertHTML("<p>&lt;</p>", "<p>&#60;</p>");
+        // actually it is not even decoded in attributes
+        assertHTML("<p foo=\"&lt;\">text</p>", "<p foo=\"&#60;\">text</p>");
+        // same if hexadecimal
+        assertHTML("<div>&lt;</div>", "<div>&#x003C;</div>");
+    }
+
+    /**
+     * Ensure that &gt; in text nodes is not decoded even if it is encoded as character entity.
+     */
+    @Test
+    void replaceClosingBracesCharacterEntityByNamedEntity()
+    {
+        assertHTML("<p>&gt;</p>", "<p>&#62;</p>");
+        // it is also not decoded in attributes (where it must not be decoded)
+        assertHTML("<p foo=\"&gt;\">text</p>", "<p foo=\"&#62;\">text</p>");
+        // same if hexadecimal
+        assertHTML("<div>&gt;</div>", "<div>&#x003E;</div>");
+    }
+
+    /**
+     * Ensures that &gt; in text nodes is encoded even if it is not encoded in the original text
+     */
+    @Test
+    void replaceBrokenExplicitClosingBraceByNamedEntity()
+    {
+        assertHTML("<p>&gt;</p>", "<p>></p>");
+        // however if a closing brace is inside an attribute, that is illegal html, and the html cleaner
+        // decides to move it to the next text node, where it is encoded (which is a legitimate decision)
+        // if this changes to something different, but also acceptable, feel free to update the test
+        assertHTML("<p foo=\"abc\">def\"&gt;text</p>", "<p foo=\"abc>def\">text</p>");
+    }
+
+    /**
+     * Test how unencoded opening angled braces aka greater-than sign are handled.
+     */
+    @Test
+    void replaceBrokenExplicitOpeningBraceByNamedEntity()
+    {
+        // ensures that &lt; in text nodes is encoded even if it is not encoded in the original text
+        assertHTML("<p>&lt;</p>", "<p><</p>");
+        // also, if it is surrounded by whitespace
+        assertHTML("<p>abc &lt; def</p>", "<p>abc < def</p>");
+        // however it does not, if it is followed by text; instead the brace and the following text are scrapped
+        assertHTML("<p>abc</p>", "<p>abc<def</p>");
+        assertHTML("<p>abc</p>", "<p>abc<def; and here we can have an much text as we want</p>");
+        // if an opening brace is inside an attribute, that is marginally ok, but the html cleaner
+        // decides to scrap it and all the following content
+        // could do this:
+        // assertHTML("<p foo=\"abc&lt;def\">text</p>", "<p foo=\"abc<def\">text</p>");
+        // but instead does:
+        assertHTML("<p foo=\"abc\">text</p>", "<p foo=\"abc<def\">text</p>");
+        // if surrounded by whitespace, then instead this happens
+        // not sure if this is good or bad; if the behavior of the HTML cleaner changes, feel free to update
+        assertHTML("<p foo=\"abc\">&lt; def\"&gt;text</p>", "<p foo=\"abc < def\">text</p>");
+    }
+
     @Test
+    @Disabled("we have no special support for textarea at the moment")
```

Comment:
I have not been able to keep this test running (except with weird hacks like encoding `&#123;` to `&amp;#123;` in the input stream, which are bound to break something else.) However I think the actual behavior is acceptable, too.